src/services/TasksApi.js

```diff
@@ -28,24 +28,36 @@ const deleteTask = (id) => {
  * Get all tasks
  *
  * @param {object} filters Filters object
- * @param {Array} filters.category Tag array to filter
+ * @param {string} filters.category Category to filter
  * @returns {Promise} The request promise
  */
-const getAllTasks = (filters) => {
-  const category = filters?.category || [];
-  const queryParams = category.length ? `?category=${category}` : '';
-  return taskApi.get(`/tasks${queryParams}`);
+const getAllTasks = ({ category } = {}) => {
+  const filters = category ? { category } : undefined;
+  return taskApi.post('/tasks/list-tasks', {
+    filters,
+    // The back-end uses 10 as the page_size default value
+    // so we need to pass a big number here to get all tasks
+    page_size: 999999,
```

Comment:
@andreluiz27 If you remove the default value of `page_size` I will remove this line from the request.